radio/src/lua/api_general.cpp

```diff
@@ -1040,23 +1040,23 @@ static int luaDefaultStick(lua_State * L)
 @param precision the precision of the sensor
  * `0 or not present` no decimal precision.
  * `!= 0` value is divided by 10^precision, e.g. value=1000, prec=2 => 10.00.
- 
+
 @param name (string) Name of the sensor if it does not yet exist (4 chars).
  * `not present` Name defaults to the Id.
  * `present` Sensor takes name of the argument. Argument must have name surrounded by quotes: e.g., "Name"
- 
+
 @retval true, if the sensor was just added. In this case the value is ignored (subsequent call will set the value)
 
 @status current Introduced in 2.2.0
 */
 static int luaSetTelemetryValue(lua_State * L)
 {
-  uint16_t id = luaL_checkinteger(L, 1);
-  uint8_t subId = luaL_checkinteger(L, 2);
-  uint8_t instance = luaL_checkinteger(L, 3);
+  uint16_t id = luaL_checkunsigned(L, 1);
+  uint8_t subId = luaL_checkunsigned(L, 2) & 0x7;
```

Comment:
Cool, first I thought that this was not needed since compiler itself take care not to assign too big value to the `telemetrySensor.subId`, but the problem would be the call to `setTelemetryValue()` because there too big `subId` would be compared with existing sensors and of course it is never found.